.storybook/src/layers/html/highlight.stories.ts

```diff
@@ -0,0 +1,174 @@
+
+  const highlightLayer = new HighlightLayer('pointhighlighter', {
+    order: 105,
+    referenceSystem,
+    layerOpacity: 0.5,
+  })
+
+  highlightLayer.onMount({ elm: container, width, height });
+
+  const zoomHandler = new ZoomPanHandler(container, (event: OnRescaleEvent) => {
+    layer.onRescale({ ...event });
+    highlightLayer.onRescale({ ...event })
+  });
+
+  zoomHandler.setBounds([0, 1000], [0, 1000]);
+  zoomHandler.adjustToSize(width, height);
+  zoomHandler.zFactor = 1;
+  zoomHandler.setTranslateBounds([-5000, 6000], [-5000, 6000]);
+  zoomHandler.enableTranslateExtent = false;
+  zoomHandler.setViewport(1000, 1000, 5000);
+
+  highlightLayer.onRescale(zoomHandler.currentStateAsEvent());
+
+  const slider = createSlider((event: any) => onUpdate(event, { rescaleEvent: zoomHandler.currentStateAsEvent(), layer: highlightLayer }), { width, min: 0, max: referenceSystem.length});
+
+  root.appendChild(container);
+  root.appendChild(slider);
+  root.appendChild(createFPSLabel());
+
+  return root;
+};
+
+export const HighlightWellborepathWithController = () => {
+  const referenceSystem = new IntersectionReferenceSystem(poslog, defaultOptions);
+
+  const layer = new WellborepathLayer('wellborepath', { order: 3, strokeWidth: '5px', stroke: 'red', referenceSystem });
+
+  const root = createRootContainer(width);
+  const container = createLayerContainer(width, height);
+
+  const controller = new Controller({
+    referenceSystem,
+    axisOptions,
+    scaleOptions,
+    container,
+  });
+
+  const highlightLayer = new HighlightLayer('pointhighlighter', {
+    order: 105,
+    referenceSystem,
+    layerOpacity: 0.5,
+  })
+
+  controller.addLayer(layer).addLayer(highlightLayer);
+
+  controller.setBounds([0, 1000], [0, 1000]);
+  controller.adjustToSize(width, height);
+  controller.setViewport(1000, 1000, 5000);
+
+  // external event that calls the rescale event the highlighting should change
+  const slider = createSlider((event: any) => onUpdate(event, { rescaleEvent: controller.currentStateAsEvent, layer: highlightLayer }), { width, min: 0, max: controller.referenceSystem.length});
+
+  root.appendChild(container);
+  root.appendChild(slider);
+  root.appendChild(createFPSLabel());
+
+  return root;
+};
+
+class HighlightLayer extends HTMLLayer {
+  elements: Selection<HTMLElement, any, null, undefined>[] = [];
+
+  onMount(event: OnMountEvent): void {
+    super.onMount(event);
+    this.addHighlightElement('wellborepath');
+  }
+
+  onRescale(event: OnRescaleEvent): void {
+    super.onRescale(event);
+    const elm = this.elements[0];
+    if (this.referenceSystem) {
+
+      // returns coords in [displacement, tvd]
+      const coords = this.referenceSystem.project(event.md || 0);
+
+      // screen coords inside the container
+      const newX = event.xScale(coords[0]);
+      const newY = event.yScale(coords[1]);
+
+      elm.style('left', `${newX - POINTOFFSET}px`);
+      elm.style('top', `${newY - POINTOFFSET}px`);
```

Comment:
Isn't this something that can be derived from the size if the element and should be defined as such?